Feature/script background.js 추가

jiwoo-choi opened the pull request:
탭이 변환되는걸 항시 체크하는 background.js 부분을 추가하였습니다.
background.js에서는 탭이 활성화하는걸 감지하고, fmkorea.com 탭이 활성화되면 지금까지의 업데이트사항을 다시한번 적용합니다.
다른탭에서 키워드 업데이트 => 다시 탭으로 돌아왔을때 업데이트사항이 적용되지 않은 부분을 대비한 부분입니다.
* 이 PR은 기존 디자인커밋 위에서 작업했습니다.

jiwoo-choi committed:
20200229 UI update : CSS update (font, color, size, class name)

jiwoo-choi committed:
20200301 completely develop for features for adding chips or removing chips. [ pull request 본]
- StorageController.js : adding functions for removing chips.
- style.css : css for chips and layouts which uses flex box.
- popup.js : change adding and removing logics using anymonous functions.
- bugfix : fixing conflicts in popup.html with previous version to fix "outside module" error.

jiwoo-choi committed:
20200305 로직 스크립트 업데이트
- background.js : 다른탭에있을때 추가하고 => 다시 fmkorea.com탭으로 돌아왔을때 업데이트사항 적용할 수 있도록하는 background.js 로직 추가.
- 나머지 소스 코드 큰 수정사항은 없습니다. (CSS 스타일 등 마이너 수정)

githubmaster1325 commented on popup.js:
```diff
@@ -31,27 +31,21 @@ async function viewDidLoad(){
+	//remove => restore?
 	sendButton.addEventListener("click", async () => {
-		await stoargeController.saveKeywordFor(input.value);
+		await stoargeController.saveKeywordFor(input.value)
```
코딩 스타일에 위배가 됩니다

한 명령어의 끝에는 ; 를 붙인다.

githubmaster1325 commented on contentMain.js:
```diff
@@ -30,11 +31,15 @@ export function contentMain() {
 
 	chrome.runtime.onMessage.addListener(
 		async function(request, sender, sendResponse) {
+			alert('hey');
```
호출이 잘 되는지 테스트 목적이라면 끝나고 지웠으면 좋겠습니다.

githubmaster1325 commented on StorageController.js:
```diff
@@ -34,13 +35,14 @@ export default class StorageController {
             } else {
                 let currentKeywords = await this.getKeywordList();
                 currentKeywords.push(key);
+                var tempDictionary = {};
```
전부 let, const 를 썼는데 여기서만 var를 쓴 이유가 있을까요?

githubmaster1325 commented on StorageController.js:
```diff
@@ -52,4 +54,25 @@ export default class StorageController {
+            if (currentKeywords.length < index) {
+                reject(OUTOF_INDEX);
+            } else {
+                var tempDictionary = {};
```
위와 동일한게 var 를 쓴 이유가 있을까요?

githubmaster1325 commented on StorageController.js:
```diff
@@ -52,4 +54,25 @@ export default class StorageController {
+                var tempDictionary = {};
+                currentKeywords.splice(index,1);
+                tempDictionary[this.keyword] = currentKeywords
+                chrome.storage.sync.set(tempDictionary, function() { 
```
끝에 ; 가 없네요.

githubmaster1325 commented on popup.html:
```diff
@@ -5,14 +5,29 @@
+    <script type="module" src="popup.js"></script>
+
+    <div class="flex-container wrap" id="chipArea">
+    <!--
```
테스트 용도의 주석은 끝난 후 삭제바랍니다.

githubmaster1325 commented on popup.js:
```diff
@@ -1,31 +1,52 @@
+		input.value = "";
+	})
+	/*
+	sendButton.addEventListener("click", async () => {
```
이 이벤트리스터가 위의 함수로 바뀐거 같은데, 필요가 없다면 삭제하면 어떨까요?

아래 broadcast라는 함수도 주석처리된 이 함수 내에서 사용 하는거 같은데 같이 삭제하면 될까요?

githubmaster1325 commented on popup.js:
```diff
@@ -1,31 +1,52 @@
-	addChip(input.value);
-	input.value = "";
-})
+	const updateChip = (keywordList) => {
```
이 updateChip 이라는 함수를 viewDidLoad 안에 넣은 이유는 전역변수를 선언하지 않기 위해서인가요?

객체 리터럴 방식을 이용해보는건 어떨까요!?

litain23 committed:
20200307 click 이벤트를 event delegation으로 구현

- popup.js : js파일 안에 있는 템플릿 html 제거 후 template
             event delegation으로 삭제에 대한 event 처리
- popup.html : div 를 ul 로 바꿔서 리스트로 처리

- storageController : keyword를 인덱스가 아닌 단어를 찾아서 삭제하는
방식으로 변경

jiwoo-choi commented on popup.js:
```diff
@@ -1,53 +1,63 @@
+		return target.previousElementSibling.innerText;
+	}
+}
+
```
findkeyword라고하니까 keyword searching하는 단계인것같아요.
getKeyword 등으로 바꾸는게 어떨까요?

jiwoo-choi commented on StorageController.js:
```diff
@@ -56,13 +56,12 @@ export default class StorageController {
-                var tempDictionary = {};
+                let tempDictionary = {};
                 currentKeywords.splice(index,1);
                 tempDictionary[this.keyword] = currentKeywords
```
삭제해도 좋을것같습니다. 중복이네요.

jiwoo-choi commented on StorageController.js:
```diff
@@ -71,8 +70,33 @@ export default class StorageController {
+
+    async removeKeyword(keyword) {
+        return new Promise(async function(resolve, reject) {
+            let currentKeywords = await this.getKeywordList();
```
이부분에서 삭제할 키워드 내용이 중복된다면 제대로 처리못해줄것같아요.
https://www.geeksforgeeks.org/sets-in-javascript/ 
Sets을 써봅시다.

jiwoo-choi reviewed (commented):
수정사항이 몇가지 있습니다. 
이 브랜치에 이어가서 작업할게요.

jiwoo-choi committed:
20200308 리뷰내용토대수정 (변수명, array append logic 수정, template사용)
- popup.html : template태그를 사용. replace로 수정하는것보다 깔끔하게 수정가능.
- popup.js : getKeywordFromAction() => 변수명 변경. (findKeyword로부터), updqteChip부분 HTML TEMPLATE 태그에 맞게 수정. try-catch문 사용이 가능한곳이면 후에 문제시 디버깅 하기 쉽도록 try-catch문 사용.
- StorageControll.js : DUPLICATE ERROR추가.
